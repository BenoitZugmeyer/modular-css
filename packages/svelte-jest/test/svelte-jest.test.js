"use strict";

const dedent = require("dedent");

const sveltejest = require("../svelte-jest.js");

describe("svelte-jest", () => {
    it("should update all {css.foo} references in the template to use their name only", () => {
        const html = dedent(`
        <link rel="stylesheet" href="./style.css" />

        <div class="{css.one}">
            <p class="{foo ? css.two : css.three}">
                CONTENT
            </p>
        </div>
        `);

        const result = sveltejest(html);

        expect(result).toMatchSnapshot();
    });
    
    it("should update all css.foo references in <script> blocks to use their name only", () => {
        const html = dedent(`
        <link rel="stylesheet" href="./style.css" />

        <script>
            export default {
                oncreate() {
                    if(true) {
                        console.log(css.foo);
                    }

                    this.set({
                        style : css.bar
                    });
                }
            };
        </script>
        `);

        const result = sveltejest(html);

        expect(result).toMatchSnapshot();
    });
});
