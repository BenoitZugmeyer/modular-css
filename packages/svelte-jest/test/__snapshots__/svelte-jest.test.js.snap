// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svelte-jest should update all {css.foo} references in the template to use their name only 1`] = `
"
<div class=\\"{\\"one\\"}\\">
    <p class=\\"{foo ? \\"two\\" : \\"three\\"}\\">
        CONTENT
    </p>
</div>"
`;

exports[`svelte-jest should update all css.foo references in <script> blocks to use their name only 1`] = `
"
<script>
    export default {
        oncreate() {
            if(true) {
                console.log(\\"foo\\");
            }

            this.set({
                style : \\"bar\\"
            });
        }
    };
</script>"
`;
